src/test/components/CallTreeSidebar.test.js

```diff
@@ -31,29 +34,76 @@ describe('CallTreeSidebar', function() {
       store.dispatch(changeSelectedCallNode(0, nodePath));
     };
 
-    return { store, funcNamesDict: funcNamesDictPerThread[0], selectNode };
-  }
-
-  it('matches the snapshots when displaying data about the currently selected node', () => {
-    const { store, selectNode, funcNamesDict: { A, B, C, D, H } } = setup();
+    const invertCallstack = () => store.dispatch(changeInvertCallstack(true));
 
     const view = mount(
       <Provider store={store}>
         <CallTreeSidebar />
       </Provider>
     );
+    return {
+      store,
+      funcNamesDict: funcNamesDictPerThread[0],
+      selectNode,
+      invertCallstack,
+      view,
+    };
+  }
+
+  it('matches the snapshots when displaying data about the currently selected node', () => {
+    const {
+      selectNode,
+      funcNamesDict: { A, B, Cjs, D, H, Ejs },
+      view,
+    } = setup();
+
     expect(view).toMatchSnapshot();
 
-    selectNode([A, B, C]);
+    // Cjs is a JS node, but has no self time, so we shouldn't see the JS engine
+    // information.
```

Comment:
In this case I would prefer to see the JS engine section but with a note that "no samples were collected with self time in this call node". But we don't have to make that change in this PR.

Reply:
Yeah I thought of this as well, and wasn't sure. I think that in the future we'll present all of this very differently too.